### PR TITLE
Move `HAControlPlanes` feature gate from gardenlet to gardener-apiserver

### LINKED ISSUE #6827: Requirements for failureTolerance "node" are not checked on Shoot creation/update
**How to categorize this issue?**

/area high-availability
/kind bug

**What happened**:
It is possible to create shoots with `failureTolerance` type `node` even when there are no seeds with this feature enabled.

_Shoot creation:_
New shoots with failure tolerance type node can be scheduled anywhere, even on seeds having HA disabled. 

_Shoot update:_
Failure tolerance of type node can be activated on shoots which are scheduled on a seed having HA disabled. 

In both cases the shoots run in a non HA setup. **This is not visible for the user**. You can just see it by checking etcd pods in control-plane.
For failure tolerance type zone on the other hand, there is a check which prevents updating the shoot (update case) or at least making them unschedulable (create case).

**What you expected to happen**:
It should not be possible to create/update shoots with failure tolerance type node if there are no suitable seeds.

**How to reproduce it (as minimally and precisely as possible)**:
Create or update shoots in a landscape without any HA seeds with this config:
```
...
spec:
  controlPlane:
    highAvailability:
      failureTolerance:
        type: node
...
```

**Anything else we need to know?**:
I found two places where the availability of `failureTolerance` type `node` on the seed is not checked
- gardener-admission-controller: [ref](https://github.com/gardener/gardener/blob/3c02b2b99184f6789306965256e940a2da71d052/plugin/pkg/shoot/validator/admission.go#L360)
- gardener-scheduler: [ref](https://github.com/gardener/gardener/blob/3c02b2b99184f6789306965256e940a2da71d052/pkg/scheduler/controller/shoot/reconciler.go#L222)

This list might not be complete.

**Environment**:

- Gardener version: v1.57.0

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area high-availability
/kind bug

**What this PR does / why we need it**:
This PR moves `HAControlPlanes` feature gate from `gardenlet` to `gardener-apiserver`.
It is more robust when `HAControlPlanes` feature gate is controlled on `gardener-apiserver` level. Otherwise it might happen, that `shoot`s in an high availability configuration are created by users, even when the landscape is not supporting this scenario yet.

When `HAControlPlanes` feature gate is disabled `gardener-apiserver` drops `HighAvailability` related fields on shoot creation and update if the existing shoot does not have a `HighAvailability` configuration yet.
When `HAControlPlanes` feature gate is enabled the fields are not dropped.

**Which issue(s) this PR fixes**:
Fixes #6827

**Special notes for your reviewer**:

**Release note**:

```breaking operator
`HAControlPlanes` feature gate is added to `gardener-apiserver` and removed from `gardenlet`.
```
```feature operator
`HAControlPlanes` feature gate controls if it is possible to create shoots with a `HighAvailability` configuration in the landscape.
```

### CODE REVIEW

File: skaffold.yaml [hunk 1]
```diff
@@ -519,6 +519,15 @@ deploy:
         helm: {}
       valuesFiles:
       - example/gardener-local/controlplane/values.yaml
+profiles:
+- name: ha
+  activation:
+  - env: HA_MODE=single-zone|multi-zone
```
Comment (rfranzke) [inline skaffold.yaml:525]:
```suggestion
```

FYI: No longer needed after https://github.com/gardener/gardener/pull/6914

**Review by rfranzke — COMMENTED** [review]
Nice, well done!

/lgtm
/cc @timuthy for approval

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Review by timuthy — COMMENTED** [review]
/lgtm

**Pull request merged**